core/classes/IssueAttachmentTimelineEvent.class.php

```diff
@@ -0,0 +1,70 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Timeline event class for file attachment operations.
+ * @copyright Copyright 2014 MantisBT Team - mantisbt-dev@lists.sourceforge.net
+ * @link http://www.mantisbt.org
+ * @package MantisBT
+ */
+
+/**
+ * Timeline event class for file attachment operations.
+ *
+ * @package MantisBT
+ * @subpackage classes
+ */
+class IssueAttachmentTimelineEvent extends TimelineEvent {
+	private $issue_id;
+	private $filename;
+	private $add;
+
+	/**
+	 * @param integer $p_timestamp Timestamp representing the time the event occurred.
+	 * @param integer $p_user_id   An user identifier.
+	 * @param integer $p_issue_id  A issue identifier.
+	 * @param string  $p_filename  Attachment's file name.
+	 * @param boolean $p_add       True if adding, false if removing attachment.
+	 */
+	public function __construct( $p_timestamp, $p_user_id, $p_issue_id, $p_filename, $p_add ) {
```

Comment:
Why not get the event type instance of an add flag? i.e. FILE_ADDED and FILE_DELETED?

Reply:
You mean instead of the $p_add boolean ?

Reply:
Yes

Reply:
Fixed in 8b1e0080745f4dd22c1b0f8dfe80efb0136f8847